### PR TITLE
DOC: Example on how to use np.lib.tracemalloc_domain.

### PR DESCRIPTION
**Pull request by JulesKouatchou**
- We provide a sample code showing how how to use `np.lib.tracemalloc_domain'. 
- Addresses https://github.com/numpy/numpy/issues/14766.

### CODE REVIEW

**Comment by mattip** [comment]
I think the place to document this is in the [data-memory](https://github.com/numpy/numpy/blob/main/doc/source/reference/c-api/data_memory.rst) page, and that this directory can be removed.

**Comment by JulesKouatchou** [comment]
As requested, I included the example at: 

      doc/source/reference/c-api/data_memory.rst

Please let me know if it is now fine.

File: doc/source/reference/c-api/data_memory.rst [hunk 1]
```diff
@@ -159,3 +159,87 @@ A better technique would be to use a ``PyCapsule`` as a base object:
         return NULL;
     }
     ...
+
+Memory Allocation
```
Comment (seberg) [inline doc/source/reference/c-api/data_memory.rst:163]: Can you find a better title pointing to tracing?  (also I guess we use sentence case not title case now)

The readme can stay with a single link to here, or just remove the entire folder.  Duplicating things doesn't seem useful.

Comment (JulesKouatchou) [reply doc/source/reference/c-api/data_memory.rst]: @seberg I changed the title of the section to include "Example of memory tracing with ``np.lib.tracemalloc_domain``". I hope it is enough. Thanks.

File: tools/allocation_tracking/README.md [hunk 2]
```diff
@@ -1,7 +1,86 @@
-Note that since Python 3.6 the builtin tracemalloc module can be used to
-track allocations inside numpy.
-Numpy places its CPU memory allocations into the `np.lib.tracemalloc_domain`
-domain.
-See https://docs.python.org/3/library/tracemalloc.html.
+
```
Comment (mattip) [inline tools/allocation_tracking/README.md:1]: Please delete this file. It is deprecated and not needed any more.

**Comment by mattip** [comment]
Thanks @JulesKouatchou

**Pull request merged**